scripts/poe/lib/utils.js

```diff
@@ -10,17 +10,25 @@ module.exports = {
   error: (msg) => {
     console.log(`${chalk.red('Error: ')}${msg}`);
   },
-  extractPlaceholders: (file) => {
+  extractPlaceholders: (file, onlyPlaceholders = false) => {
     const content = fs.readFileSync(file, 'utf8');
 
+    if (onlyPlaceholders) {
+      return content
+              .toString()
+              .match(/{{.+?}}/g)
+              .sort()
+              .filter((el,i,a) => i === a.indexOf(el));
+    }
+
     const result = {};
     content
       .toString()
       .split('\n')
       .forEach((line, index) => {
         const placeholders = line.match(/{{.+?}}/g);
         if (placeholders) {
-          placeholders.sort();
+          placeholders.sort().filter((el,i,a) => i === a.indexOf(el));
```

Comment:
There's quite a lot of duplicate code between this and the `onlyPlaceholders` branch - can you see a way to reuse the code?